ports/raspberrypi/bindings/rp2pio/StateMachine.c

```diff
@@ -42,6 +42,15 @@
 #include "py/runtime.h"
 #include "supervisor/shared/translate.h"
 
+static void update_typecode_maybe(mp_buffer_info_t *bufinfo, mp_obj_t typecode_maybe) {
+    if (typecode_maybe != MP_ROM_NONE) {
+        // no explicit error handling is OK here as:
+        // * a non-string throws in mp_obj_str_get_str
+        // * An empty string is nul-terminated, which will lead to 'bad typecode' later in mp_binary_get_size
```

Comment:
`nul` or `null`?

Reply:
Meaning the special byte with value 0 that terminates a C-string. I can write it however you prefer.

Reply:
either `NUL` or `null` is fine. I thought `nul` was a typo.